cfme/tests/automate/custom_button/test_infra_objects.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+import fauxfactory
+
+from widgetastic_patternfly import Dropdown
+
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_TYPE
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.log import logger
+
+
+pytestmark = [
+    pytest.mark.tier(2),
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE),
+]
+
+INFRA_OBJECTS = ["PROVIDER"]
+
+DISPLAY_NAV = {
+    "Single entity": ["Details"],
+    "List": ["All"],
+    "Single and list": ["All", "Details"],
+}
+
+
+@pytest.fixture(
+    params=INFRA_OBJECTS, ids=[obj.capitalize() for obj in INFRA_OBJECTS], scope="module"
+)
+def button_group(appliance, request):
+    collection = appliance.collections.button_groups
+    button_gp = collection.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        type=getattr(collection, request.param),
+    )
+    yield button_gp, request.param
+    button_gp.delete_if_exists()
+
+
+@pytest.fixture()
+def setup_obj(button_group, provider):
+    """ Setup object for specific custom button object type.
+    """
```

Comment:
This is a one line comment
""" Setup object for specific custom button object type."""